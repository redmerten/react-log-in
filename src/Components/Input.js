/**
 * Created by AndreaMerten on 3/14/18.
 */

import React from 'react'

const Input = ({label, value, onChange, style}) =>{
  return(
    <div style={styles.div}>
      <label htmlFor="email" style={styles.label}>{label}</label>
      <input
        type="text"
        value={value}
        onChange={onChange}
        style={style}
      />
    </div>
  )
}

export default Input

const styles={
  div:{
    display:'flex',
    margin:'3% 0%',
    justifyContent:'space-between',
    alignItems:'center'
  },
  label:{
    fontSize:'150%',
    margin:'0% 2%',
  }
}