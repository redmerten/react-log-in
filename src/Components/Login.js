/**
 * Created by AndreaMerten on 1/17/18.
 */
import React, {Component} from 'react'
import {connect} from 'react-redux'

import {storeLogin} from '../Actions'
import Input from './Input'

export class Login extends Component{
  state={
    email:'',
    password:'',
    valid:false,
    invalidEntry: false
  }

  handleEmailChange=(event)=>{
    this.setState({email:event.target.value.trim()})
  }

  handlePasswordChange=(event)=>{
    this.setState({password:event.target.value.trim()})
  }

  handleSubmit=(event)=>{
    event.preventDefault()
    const {email, password} = this.props.login
    //validate
    if (this.state.email === email
        &&
        this.state.password === password){
      this.setState({valid:true})
    }
    else{
      alert('Email or Password Incorrect')
      this.setState({valid:false, email:'', password:''})
    }
  }


  render() {
    let {login} = this.props
    console.log(login)
    return (
      <form onSubmit={this.handleSubmit} style={styles.form}>
        <h2 style={styles.h2}>Login</h2>
        <Input
          value={this.state.email}
          onChange={(event)=>this.handleEmailChange(event)}
          style={styles.field}
          label='Email'
        />
        <Input
          value={this.state.password}
          onChange={(event)=>this.handlePasswordChange(event)}
          style={styles.field}
          label='Password'
          />

        <button
          type="submit"
          style={styles.button}
          onSubmit={(e)=>this.handleSubmit(e)}
        >
          Submit
        </button>
        {this.state.valid
          ?
          <h2 style={styles.h2}>You successfully logged in</h2>
          :
          <div/>
        }
      </form>
    )

  }
}

const mapStateToProps = (state) =>{
  return state
}

export default connect(mapStateToProps, {storeLogin})(Login)



const styles={
  form:{
    display:'flex',
    //width:'50%',
    margin:'5% 38%',
    flexDirection:'column',
    justifyContent:'flex-begin',
    //alignItems:'center'
  },
  h2:{
    alignSelf:'center'
  },
  div:{
    display:'flex',
    margin:'3% 0%',
    justifyContent:'space-between',
    alignItems:'center'
  },
  label:{
    fontSize:'150%',
    margin:'0% 2%',
  },
  field:{
    height:'30px',
    width:'250px',
    fontSize:'125%',
    textIndent:'5px',
    margin:'0% 2%'
  },
  button:{
    height:'40px',
    width:'250px',
    margin:'3% 0%',
    alignSelf:'center',
    backgroundColor:'#709dff',
    borderRadius:'5%',
    fontSize:'150%',

  }

}
