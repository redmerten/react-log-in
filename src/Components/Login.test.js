/**
 * Created by AndreaMerten on 3/14/18.
 */

import React, {Component} from 'react'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'

import {Login} from './Login'
import Input from './Input'

configure({ adapter: new Adapter() });
const props = {
  storeLogin : jest.fn(() => ({}))
}

describe('<Login/>', () => {
  const wrapper = shallow(<Login {...props}/>)

  it('renders a form element', () => {
    expect(wrapper.find(<form/>))
  })

  it('renders a snapshot', () => {
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})

