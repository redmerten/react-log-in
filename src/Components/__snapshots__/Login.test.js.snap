// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders a snapshot 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-begin",
      "margin": "5% 38%",
    }
  }
>
  <h2
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    Login
  </h2>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "3% 0%",
      }
    }
  >
    <label
      htmlFor="email"
      style={
        Object {
          "fontSize": "150%",
          "margin": "0% 2%",
        }
      }
    >
      Email
    </label>
    <input
      onChange={[Function]}
      style={
        Object {
          "fontSize": "125%",
          "height": "30px",
          "margin": "0% 2%",
          "textIndent": "5px",
          "width": "250px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "3% 0%",
      }
    }
  >
    <label
      htmlFor="email"
      style={
        Object {
          "fontSize": "150%",
          "margin": "0% 2%",
        }
      }
    >
      Password
    </label>
    <input
      onChange={[Function]}
      style={
        Object {
          "fontSize": "125%",
          "height": "30px",
          "margin": "0% 2%",
          "textIndent": "5px",
          "width": "250px",
        }
      }
      type="text"
      value=""
    />
  </div>
  <button
    onSubmit={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#8e8e8e",
        "borderRadius": "5%",
        "fontSize": "150%",
        "height": "40px",
        "margin": "3% 0%",
        "width": "250px",
      }
    }
    type="submit"
  >
    Submit
  </button>
  <div />
</form>
`;
